debezium-core/src/main/java/io/debezium/relational/history/KafkaDatabaseHistory.java

```diff
@@ -177,27 +179,25 @@ protected void storeRecord(HistoryRecord record) throws DatabaseHistoryException
     }
 
     @Override
-    protected void recoverRecords(Tables schema, DdlParser ddlParser, Consumer<HistoryRecord> records) {
+    protected void recoverRecords(Consumer<HistoryRecord> records) {
         try (KafkaConsumer<String, String> historyConsumer = new KafkaConsumer<>(consumerConfig.asProperties());) {
             // Subscribe to the only partition for this topic, and seek to the beginning of that partition ...
             logger.debug("Subscribing to database history topic '{}'", topicName);
             historyConsumer.subscribe(Collect.arrayListOf(topicName));
 
-            // Explicitly seek to the beginning of all assigned topic partitions ...
-            logger.debug("Seeking Kafka consumer to beginning of all assigned topic partitions");
-            historyConsumer.seekToBeginning(Collections.emptyList());
-
             // Read all messages in the topic ...
             int remainingEmptyPollResults = this.recoveryAttempts;
-            Map<Long, Long> offsetsByPartition = new HashMap<>();
-            while (remainingEmptyPollResults > 0) {
+            long lastProcessedOffset = -1;
+            long endOffset = getEndOffsetOfDbHistoryTopic(historyConsumer);
+
+            // read the topic until the end
+            while (lastProcessedOffset < endOffset - 1 && remainingEmptyPollResults > 0) {
```

Comment:
Wondering whether we should even keep the check for `remainingEmptyPollResults > 0` or not rather remove it and repeat until we've read until the end offset.

Reply:
Ok, so I've done that in the last commit. IIUC, that's how it should be.